main_test.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"net/url"
+	"testing"
+	"time"
+
+	prometheus "github.com/prometheus/client_model/go"
+)
+
+// Tests if retries are working for tenant registrations.
+// To test this, a tenant with wrong configuration needs to be present at /test/conf/tenants.yaml.
+func TestOnboardTenants(t *testing.T) {
+	cfg := config{}
+
+	cfg.server.listen = "0.0.0.0:8443"
+	cfg.server.listenInternal = "0.0.0.0:8448"
+
+	u, err := url.ParseRequestURI("http://127.0.0.1:3100")
+
+	if err != nil {
+		t.FailNow()
+	}
+
+	cfg.logs.readEndpoint = u
+
+	u, err = url.ParseRequestURI("http://127.0.0.1:3100")
+
+	if err != nil {
+		t.FailNow()
+	}
+
+	cfg.logs.writeEndpoint = u
+
+	u, err = url.ParseRequestURI("http://127.0.0.1:9091")
+
+	if err != nil {
+		t.FailNow()
+	}
+
+	cfg.metrics.readEndpoint = u
+
+	u, err = url.ParseRequestURI("http://127.0.0.1:19291")
+
+	if err != nil {
+		t.FailNow()
+	}
```

Comment:
Suggest to skip checking errors for `net/url` functionality and strip down this test case.